Fixed CV view

JoshuaStorm opened the pull request:
See #3974 
This time on correct branch.

"Fixes #3878

Transition between 2D and CV views has a bit of a jump due to leaving orthographic view, similar to the jerkiness of 2D view entering orthographic view. I will look into fixing that as well, but that may be worth a separate PR."

pjcozzi commented:
@lilleyse do you plan to review this?

lilleyse commented:
Oh yes, somehow this slipped by. I'll review soon.

lilleyse commented:
The 3D to CV transition looks good, though maybe the atmosphere should fade out as the Earth rotates down.

The 2D to CV is still a little weird. Maybe the perspective shift should happen first, and then rotate the globe.

JoshuaStorm commented:
Updated. At this point, 2D to CV is a considerably better transition as I switched to shift perspective first. However, this results in a weird jerk forward before the transition and I can't quite find what is causing this.

pjcozzi commented:
@lilleyse what is the latest?

lilleyse commented:
I'm also noticing the jump from 2D to CV. Maybe it's because the initial CV view is titled, but should be straight when switching from 2D.

JoshuaStorm commented:
I have come upon a somewhat hacky fix: the issue seems to be that from orthographic view the scene.camera.position does not seem to reflect what the viewer actually sees.

A fix for this was just a hardcoded fine tuning of the z position of the camera. This results in a fluid transition _if_ the user allowed 2D view to go fully into orthographic view. Otherwise, there is a slight jump you will see.

lilleyse commented:
The effect looks good, but I am occasionally seeing crashes if I go from 2D to CV.

lilleyse commented:
Actually those seem to be unrelated.

pjcozzi commented:
What's the plan here?  Can we get this in shape for 1.23 or just bail on it?
